specification/kubernetesruntime/KubernetesRuntime.Management/storageclass.tsp

```diff
@@ -0,0 +1,276 @@
+  @doc("RWX StorageClass")
+  type: SCType.RWX;
+
+  @doc("The backing storageclass used to create new storageclass")
+  backingStorageClassName: string;
+}
+
+@doc("The properties of Blob StorageClass")
+model BlobStorageClassTypeProperties extends StorageClassTypeProperties {
+  @doc("Blob StorageClass")
+  type: SCType.Blob;
+
+  @doc("Azure Storage Account Name")
+  azureStorageAccountName: string;
+
+  @doc("Azure Storage Account Key")
+  @secret
+  azureStorageAccountKey: string;
+}
+
+@doc("The action to take when a NFS volume is deleted")
+enum NfsDirectoryActionOnVolumeDeletion {
+  @doc("When the volume is deleted, delete the directory")
+  Delete,
+
+  @doc("When the volume is deleted, retain the directory")
+  Retain,
+}
+
+@doc("The properties of NFS StorageClass")
+model NfsStorageClassTypeProperties extends StorageClassTypeProperties {
+  @doc("NFS StorageClass")
+  type: SCType.NFS;
+
+  @doc("NFS Server")
+  server: string;
+
+  @doc("NFS share")
+  share: string;
+
+  @doc("Sub directory under share. If the sub directory doesn't exist, driver will create it")
+  subDir?: string;
+
+  @doc("Mounted folder permissions. Default is 0. If set as non-zero, driver will perform `chmod` after mount")
+  mountPermissions?: string;
+
+  @doc("The action to take when a NFS volume is deleted. Default is Delete")
+  onDelete?: NfsDirectoryActionOnVolumeDeletion;
+}
+
+@doc("The properties of SMB StorageClass")
+model SmbStorageClassTypeProperties extends StorageClassTypeProperties {
+  @doc("SMB StorageClass")
+  type: SCType.SMB;
+
+  @doc("SMB Source")
+  source: string;
+
+  @doc("Sub directory under share. If the sub directory doesn't exist, driver will create it")
+  subDir?: string;
+
+  @doc("Server username")
+  username?: string;
+
+  @doc("Server password")
+  @secret
+  password?: string;
+
+  @doc("Server domain")
+  domain?: string;
+}
+
+@doc("The model for update a storageClass")
+model StorageClassTypePropertiesUpdate {
+  ...OptionalProperties<OmitProperties<RwxStorageClassTypeProperties, "type">>;
+  ...OptionalProperties<OmitProperties<BlobStorageClassTypeProperties, "type">>;
+  ...OptionalProperties<OmitProperties<NfsStorageClassTypeProperties, "type">>;
+  ...OptionalProperties<OmitProperties<
+    SmbStorageClassTypeProperties,
+    "type" | "subDir"
+  >>;
+}
+
+@doc("The model for updating storageClass properties")
+model StorageClassPropertiesUpdate {
+  ...Azure.ResourceManager.Foundations.ResourceUpdateModelProperties<
+    StorageClassResource,
+    StorageClassProperties
+  >;
+
+  @doc("New storage class type of storageClass")
+  typeProperties?: StorageClassTypePropertiesUpdate;
+}
+
+@doc("The model for updating a storageClass")
+model StorageClassResourceUpdate {
+  @doc("The properties of StorageClass")
+  properties?: StorageClassPropertiesUpdate;
+}
+
+@doc("A StorageClass resource")
+model StorageClassResource is ExtensionResource<StorageClassProperties> {
```

Comment:
>ExtensionResource [](http://example.com/codeflow?start=30&length=17)

What all types can it extend? You should try to document the expected usage scenarios

Reply:
Should this information be written to the API doc? I think it should be enough to explain it in our product documentation (not completed yet)